controllers/configurationpolicy_controller.go

```diff
@@ -364,6 +361,104 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 	relatedObjects := []policyv1.RelatedObject{}
 	parentStatusUpdateNeeded := false
 
+	// Collect template information for processing:
+	// - Object name, if specified
+	// - Object namespace, if specified
+	// - Whether object is namespaced
+	//
+	// Also determine whether a namespace query is needed for this policy:
+	//   If a namespaceSelector is present AND objects are namespaced without a namespace specified
+	templateObjs := make([]struct {
+		name         string
+		namespace    string
+		isNamespaced bool
+	}, len(plc.Spec.ObjectTemplates))
+	selectedNamespaces := []string{}
+	queryNamespaces := false
+
+	for idx, objectT := range plc.Spec.ObjectTemplates {
+		var blob interface{}
+		if jsonErr := json.Unmarshal(objectT.ObjectDefinition.Raw, &blob); jsonErr != nil {
+			log.Error(jsonErr, "Could not unmarshal data from JSON")
+
+			return
+		}
+
+		var unstruct unstructured.Unstructured
+		unstruct.Object = make(map[string]interface{})
+		unstruct.Object = blob.(map[string]interface{})
+
+		templateObjs[idx].isNamespaced = r.isObjectNamespaced(&unstruct, true)
+
+		if md, ok := unstruct.Object["metadata"]; ok {
+			metadata := md.(map[string]interface{})
+
+			if objNs, ok := metadata["namespace"]; ok {
+				templateObjs[idx].namespace = objNs.(string)
+			} else if templateObjs[idx].isNamespaced {
+				log.V(2).Info(
+					"metadata.namespace not found or could not be retrieved from the object template", "index", idx,
+				)
+			}
+
+			if objName, ok := metadata["name"]; ok {
+				templateObjs[idx].name = objName.(string)
+			} else {
+				log.V(2).Info(
+					"metadata.name not found or could not be retrieved from the object template", "index", idx,
+				)
+			}
+		}
+
+		if queryNamespaces ||
```

Comment:
I find this easier to read:
```suggestion
		if !queryNamespaces &&
```

Reply:
You could also just remove the `queryNamespaces` check all together. I wasn't sure if you included that for a slight performance improvement.

Reply:
Ah, right. That would be clearest. Yeah, I think I was but I think the performance impact would be minimal.